pkg/component/watch.go

```diff
@@ -109,22 +109,37 @@ func addRecursiveWatch(watcher *fsnotify.Watcher, path string, ignores []string)
 
 var UserRequestedWatchExit = fmt.Errorf("safely exiting from filesystem watch based on user request")
 
+// WatchParameters is designed to hold the controllables and attributes that the watch function works on
+type WatchParameters struct {
+	// Name of component that is to be watched
+	ComponentName string
+	// Name of application, the component is part of
+	ApplicationName string
+	// The path to the source of component(local or binary)
+	Path string
+	// List/Slice of files/folders in component source, the updates to which need not be pushed to component deployed pod
+	FileIgnores []string
+	// Custom function that can be used to push detected changes to remote pod. For more info about what each of the parameters to this function, please refer, pkg/component/component.go#PushLocal
+	WatchHandler func(*occlient.Client, string, string, string, io.Writer, []string) error
+	// This is a channel added to signal readiness of the watch command to the external channel listeners
+	StartChan chan string
```

Comment:
could those channels be just a `boolean`?

Reply:
Ack